dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,784 @@
+
+                    lmbda = (
+                        scipy.constants.c * 1e-6 / bdata.freq[:, np.newaxis, np.newaxis]
+                    )
+
+                    # Loop over polarizations
+                    for ii, pol in enumerate(pols):
+
+                        self.logger.info("Processing Pol %s" % pol)
+
+                        pvis = bdata.vis[:, prod[pol], :]
+                        pvar = bvar[:, prod[pol], :]
+                        pflag = bflag[:, prod[pol], :]
+                        pcnt = cnt[np.newaxis, prod[pol], :]
+                        pscale = scale[pol][np.newaxis, :, np.newaxis]
+
+                        fringestop_phase = tools.fringestop_phase(
+                            ha,
+                            lat,
+                            src_dec,
+                            dist[pol][np.newaxis, :, 0, np.newaxis] / lmbda,
+                            dist[pol][np.newaxis, :, 1, np.newaxis] / lmbda,
+                        )
+
+                        vis[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag * pvis * fringestop_phase, axis=1
+                        )
+                        var[freq_sel, ii, :] += np.sum(
+                            (pscale * pcnt) ** 2 * pflag * pvar, axis=1
+                        )
+                        counter[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag, axis=1
+                        )
+
+                    self.logger.info(
+                        "Took %0.1f seconds to process this block."
+                        % (time.time() - t0,)
+                    )
+
+                    del bdata
+                    gc.collect()
+
+                # Normalize
+                inv_counter = tools.invert_no_zero(counter)
+                vis *= inv_counter
+                var *= inv_counter ** 2
+
+                ra = np.degrees(np.unwrap(ra))
+                ha = ra - np.degrees(src_ra)
+
+                # Fit response to model
+                fwhm = np.zeros((nfreq, npol), dtype=np.float32)
+                for ii in range(npol):
+                    fwhm[:, ii] = cal_utils.guess_fwhm(
+                        data.freq, pol=pols[ii][0], dec=src_dec, sigma=True
+                    )
+
+                flag = counter > 0.0
+
+                fitter = cal_utils.FitGaussAmpPolyPhase(poly_deg_phi=self.poly_deg_phi)
+                fitter.fit(ha, vis.real, np.sqrt(var), width=fwhm, absolute_sigma=True)
+                resid = vis.real - fitter.predict(ha)
+                resid_rms = np.std(resid, axis=-1)
+
+                if self.peak_type == "max_amp":
+                    # To have one slice in time
+                    ha_max = np.nanmedian(fitter.peak())
+                elif self.peak_type == "zero_ha":
+                    ha_max = 0.0
+                else:
+                    raise ValueError("Peak type not recognized")
+
+                peak_flux = fitter.predict(ha_max)
```

Comment:
this can also be done in setup (see last comment)

Reply:
For `self.peak_type == "max_amp"`, it needs to locate ha corresponding to peak flux. Moving it to `setup` may be tricky.

Reply:
Oh yeah, my mistake.